enterprise/tpa_pipeline.py

```diff
@@ -0,0 +1,153 @@
+"""
+Module provides elements to be used in third-party auth pipeline
+"""
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+
+try:
+    from third_party_auth.provider import Registry
+except ImportError:
+    Registry = None
+
+from enterprise.models import EnterpriseCustomer, UserDataSharingConsentAudit
+from enterprise.utils import NotConnectedToEdX
+
+from social.pipeline import partial
+
+class MultipleCustomersFound(Exception):
+    """
+    Raise this exception if there are multiple customers tied to an IdP.
+    """
+
+
+def get_enterprise_customer_for_request(request):
+    """
+    Get the EnterpriseCustomer associated with a particular request.
+    """
+    if Registry is None:
+        raise NotConnectedToEdX(
+            _("This package must be installed in an EdX environment to look up third-party auth providers.")
+        )
+    pipeline = request.session.get('partial_pipeline')
+    provider = Registry.get_from_pipeline(pipeline)
+    return get_enterprise_customer_for_sso(provider)
+
+
+def get_enterprise_customer_for_running_pipeline(pipeline):
+    """
+    Get the EnterpriseCustomer associated with a running pipeline
+    """
+    if Registry is None:
+        raise NotConnectedToEdX(
+            _("This package must be installed in an EdX environment to look up third-party auth providers.")
```

Comment:
DRY: Move this repeated string to within the `NotConnectedToEdX` definition itself.